rfcs/79-minimal-fragmenting-relationships.md

```diff
@@ -0,0 +1,145 @@
+# Feature Name: `minimal-fragmenting-relationships`
+
+## Summary
+Users often run into difficulty while trying to model entity hierarchies in Bevy. The tools we currently have are not expressive enough to meet their needs. See this issue for more discussion [#3742](https://github.com/bevyengine/bevy/issues/3742).
+
+This RFC proposes a minimal design that would unblock a large amount of users while also leaving avenues for the engine to further expand these features. In short:
+- Define a new variant of `Identifier` that combines a `ComponentId` (the relationship type) and an `Entity` (the target) that can be added to an entity (the source) to model a relationship of that type between the source and the target.
+- Implement command, world and query APIs to support adding, removing and reading these new relationships.
+
+A large amount of internal refactoring has to be done in order to support this API so a significant focus will be on eliminating these technical blockers. For example:
+- Replacing internal `SparseSet` usage where `Identifier` would be used as keys as they are no longer dense.
+- Allowing archetypes to be deleted when an entity that was used as part of a pair in that archetype is despawned. This involves updating several internal caches that are used for correctness and/or safety.
+
+## Motivation
```

Comment:
This section needs to be rewritten to take into account the newly merged non-fragmenting relations.